modules/Bio/EnsEMBL/Compara/RunnableDB/HomologyAnnotation/DiamondFromFilePAF.pm

```diff
@@ -0,0 +1,98 @@
+=head1 LICENSE
+
+See the NOTICE file distributed with this work for additional information
+regarding copyright ownership.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=head1 NAME
+
+Bio::EnsEMBL::Compara::RunnableDB::HomologyAnnotation::DiamondFromFilePAF
+
+=head1 DESCRIPTION
+
+Run DIAMOND blastp and parse the output into PeptideAlignFeature objects.
+Store PeptideAlignFeature objects in the compara database
+
+Supported parameters:
+    'ref_fasta'    : the query fasta sequences already in fasta file format per genome (Mandatory)
+    'blast_db'     : the predefined and indexed diamond database name (Mandatory)
+    'blast_params' : additional blast parameters that are not the default (Optional)
+    'evalue_limit' : the minimum allowed evalue to filter results - blast/diamond parameter (Optional)
+    'genome_db_id' : the genome_db_id of the query genome (single genome in ref_fasta) (Mandatory)
+
+=cut
+
+package Bio::EnsEMBL::Compara::RunnableDB::HomologyAnnotation::DiamondFromFilePAF;
+
+use warnings;
+use strict;
+
+use Bio::EnsEMBL::Registry;
+use Bio::EnsEMBL::Compara::DBSQL::DBAdaptor;
+
+use Data::Dumper;
+
+use base ('Bio::EnsEMBL::Compara::RunnableDB::BlastAndParsePAF');
+
+sub param_defaults {
+    my ($self) = @_;
+    return {
+        %{$self->SUPER::param_defaults},
+    }
+}
+
+sub fetch_input {
+    my $self = shift;
+
+    my $output_dba = Bio::EnsEMBL::Compara::DBSQL::DBAdaptor->go_figure_compara_dba( $self->param('output_db') );
+    $self->param('output_dba', $output_dba);
+}
+
+sub run {
```

Comment:
The two runnables are indeed similar in function, but they are written quite differently. I separated them into the two different runnables wrapping `Bio::EnsEMBL::Compara::RunnableDB::BlastAndParsePAF` because I find the base is entirely horrible to go through with so many edge cases based on parameters and I have tried to avoid that. I think for this iteration it will also take too much time to merge them now, better to get out there what works if that is okay.